### PR TITLE
[front] cleanup: remove gpt3-5 model

### PR DESCRIPTION
**Pull request by tdraier**
fixes: https://github.com/dust-tt/tasks/issues/1153

## Description

Remove gpt3.5 from supported models and default assistant

## Risk

Ensure all assistants are correctly migrated before removing from supported models. Can be reverted if needed

## Deploy Plan

migrate : `npx tsx ./scripts/update_assistants_models.ts -e  --fromModel gpt-3.5-turbo --toModel gpt-4o-mini`
deploy `front`
migrate again : `npx tsx ./scripts/update_assistants_models.ts -e  --fromModel gpt-3.5-turbo --toModel gpt-4o-mini`

### CODE REVIEW

File: front/lib/api/assistant/global_agents.ts [hunk 1]
```diff
@@ -184,37 +184,6 @@ function _getHelperGlobalAgent({
   };
 }
 
-function _getGPT35TurboGlobalAgent({
```
Comment (spolu) [inline front/lib/api/assistant/global_agents.ts:187]: Unfortunatel we can't remove it entirely as past conversations will require it to render properly. We need to keep that function and have this one be moved in the `RETIRED_GLOABL_AGENTS_SID`

File: front/lib/api/assistant/global_agents.ts [hunk 2]
```diff
@@ -245,8 +214,8 @@ function _getGPT4GlobalAgent({
     scope: "global",
     userListStatus: status === "active" ? "in-list" : "not-in-list",
     model: {
-      providerId: GPT_4O_MODEL_CONFIG.providerId,
-      modelId: GPT_4O_MODEL_CONFIG.modelId,
+      providerId: GPT_4O_MINI_MODEL_CONFIG.providerId,
```
Comment (spolu) [inline front/lib/api/assistant/global_agents.ts:217]: ? We def don't want to point gpt4 to the mini model?

File: front/lib/api/assistant/global_agents.ts [hunk 3]
```diff
@@ -1057,9 +1026,6 @@ function getGlobalAgent(
         helperPromptInstance,
       });
       break;
-    case GLOBAL_AGENTS_SID.GPT35_TURBO:
```
Comment (spolu) [inline front/lib/api/assistant/global_agents.ts:1060]: need to be kept

File: types/src/front/lib/assistant.ts [hunk 4]
```diff
@@ -207,21 +205,6 @@ export const GPT_4O_MINI_MODEL_CONFIG: ModelConfigurationType = {
   toolUseMetaPrompt: OPEN_AI_TOOL_USE_META_PROMPT,
   supportsVision: true,
 };
-export const GPT_3_5_TURBO_MODEL_CONFIG: ModelConfigurationType = {
```
Comment (spolu) [inline types/src/front/lib/assistant.ts:210]: this one is ok to remove :+1:

File: front/lib/api/assistant/global_agents.ts (hunk 2 above)
Comment (tdraier) [reply front/lib/api/assistant/global_agents.ts:217]: definitely not, don't know what I did here

File: types/src/front/lib/assistant.ts (hunk 4 above)
Comment (tdraier) [reply types/src/front/lib/assistant.ts:210]: I actually need this to implement _getGPT35TurboGlobalAgent (at least I need `modelId` / `providerId` / `description` ) and model is needed in SUPPORTED_MODEL_CONFIGS to support `agent_messages_content_parser.getDelimitersConfiguration`

**Review by spolu — APPROVED** [review]
LGTM

But we have to migrate as well as deploying front right, otherwise assistant builder will fail on assistants with gpt35? Can you explicit the deploy plan accordingly? 

I think the assistant builder does not really fail but moves back to the default so it's fine to migrate deploy migrate or even deploy and migrate.

**Comment by tdraier** [comment]
> LGTM
> 
> But we have to migrate as well as deploying front right, otherwise assistant builder will fail on assistants with gpt35? Can you explicit the deploy plan accordingly?
> 
> I think the assistant builder does not really fail but moves back to the default so it's fine to migrate deploy migrate or even deploy and migrate.

yes, migration description was here : 
https://github.com/dust-tt/tasks/issues/1153#issuecomment-2288686230 . Adding this here, migrate - deploy - migrate looks fine

**Pull request merged**